appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,227 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Any, TypeVar
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+C = TypeVar("C", bound="SupportsCapabilities")
+
+
+class AdbOptionsDescriptor:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __get__(self, obj: C, cls: type[C]) -> Any:
+        if self.name == 'ADB_EXEC_TIMEOUT':
+            value = getattr(obj, 'get_capability')(self.name)
+            return None if value is None else timedelta(milliseconds=value)
+        return getattr(obj, 'get_capability')(self.name)
```

Comment:
getattr is now redundant because linter knows that the object implements the corresponding interface

Reply:
> getattr is now redundant because linter knows that the object implements the corresponding interface

I did not quite understand this. Correct me if am wrong. In the descriptor class we have to have `__get__` method as it internally calls `get_capability` method as soon as we do an attribute lookup, which in turn pulls out the value of attribute let's say `ADB_EXEC_TIMEOUT` from the internal dictionary.  So I have to ask `__get__` method to call `get_capability` method in which is implemented in `base.py`, which I am doing it through `getattr` method. I am confused as to how `getattr` is redundant in this case.

Reply:
> getattr is now redundant because linter knows that the object implements the corresponding interface

I have made this change in the code. Now instead of using `getattr` and `setattr` method I am directly calling `get_capability` and `set_capability` methods using `obj` in `__get__` and `__set__` method.